compose/ui/ui-graphics/src/skikoMain/kotlin/androidx/compose/ui/graphics/layer/SkiaGraphicsLayer.skiko.kt

```diff
@@ -292,19 +321,20 @@ actual class GraphicsLayer internal constructor() {
                         }
                     }
                 )
+                restoreCount++
             } else {
                 canvas.save()
+                restoreCount++
             }
 
-            canvas.nativeCanvas.drawPicture(it, null, null)
+            // Read the state because any changes to the state should trigger re-drawing.
+            drawState.value
```

Comment:
Shouldn't it be outside of `picture?.let {`?

Reply:
IMO It should be in a place where we actually draw. Pre-checks for invalid state might be improved in future

Reply:
when `picture` is null, we also draw, just nothing.

Imagine, we draw nothing, but then we actually created the picture => we should invalidate

Reply:
>Imagine, we draw nothing, but then we actually created the picture => we should invalidate

Actually, if we don't treat `record` as mutating state, then we shouldn't. If we treat, then we also should call `requestDraw` inside `record`

Reply:
I see that we call `updateLayerConfiguration` there, which calls `requestDraw`

Reply:
Discussed in Slack, fixed